tool/stolonctl/internal/pkg/kubernetes/status.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright (C) 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/util/node"
+)
+
+// PodStatus represents status of Kubernetes Pod
+type PodStatus struct {
+	// Pod name
+	Name string
+	// IP address of the host to which the is assigned
+	HostIP string
+	// IP address allocated to the pod
+	PodIP string
+	// Status of the pod
+	Status string
+	// ReadyContainers - number of ready container
+	ReadyContainers int
+	// TotalContainers - total number of containers in Pod
+	TotalContainers int
+	// CreationTimestamp is a timestamp representing the server time
+	// when this pod was created
+	CreationTimestamp *metav1.Time
+}
+
+// DeterminePodStatus - k8s.io/kubernetes/pkg/kubectl/resource_printer.go
+func DeterminePodStatus(pod v1.Pod) (string, int, int) {
+	restarts := 0
+	totalContainers := len(pod.Spec.Containers)
+	readyContainers := 0
+
+	reason := string(pod.Status.Phase)
+	if pod.Status.Reason != "" {
+		reason = pod.Status.Reason
+	}
+
+	initializing := false
```

Comment:
I think it's safe to start with `initializing := true` and reset it in the first case